lib/email_events/mailer.rb

```diff
@@ -23,8 +23,13 @@ def self.track_data(tracked_data_method, attrs = {})
           self.sent_email_data_class = attrs[:class] if attrs.has_key? :class
         end
 
+        def self.__track_data?
+          # track the sent email if there's either an event handler or a custom track data method
+          self.event_handler.present? || self.tracked_data_method.present?
```

Comment:
This flexibly supports strings or methods to be set for this value, so `respond_to?` wouldn't cover the former.